packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -263,12 +265,42 @@ class DeepLinksController extends DisposableController {
   final textEditingController = TextEditingController();
   final deepLinksServices = DeepLinksServices();
 
+  void addLocalFingerprint(BuildContext context, String fingerprint) async {
+    // A valid fingerprint consists of 32 pairs of hexadecimal digits separated by colons.
+    bool isValidFingerpint(String input) {
+      final RegExp pattern =
+          RegExp(r'^([0-9a-f]{31}:){5}[0-9a-f]{2}$', caseSensitive: false);
+      return pattern.hasMatch(input);
+    }
+
+    if (!isValidFingerpint(fingerprint)) {
+      await showDialog(
```

Comment:
we should trigger this from the Flutter build method so that we don't have to pass a BuildContext around.

so instead, this addLocalFingerprint method could return a bool whether the fingerprint was added successfully, and then in the UI code, if it was not successful we can show the dialog

Reply:
This makes sense, thank you!